src/main/java/uk/gov/justice/hmpps/casenotes/sync/SyncCaseNotes.kt

```diff
@@ -164,12 +167,17 @@ class SyncCaseNotes(
 
   @Transactional(readOnly = true)
   fun resendEvents(request: ResendPersonCaseNoteEvents) {
-    noteRepository.findAllByIdIn(request.uuids)
+    noteRepository.findAll(request.asSpecification())
       .map { it.createEvent(UPDATED, sourceOverride = Source.DPS) }
       .forEach(eventPublisher::publishEvent)
   }
 }
 
+private fun ResendPersonCaseNoteEvents.asSpecification(): Specification<Note> = listOfNotNull(
+  if (uuids.isNotEmpty()) idIn(uuids) else null,
+  createdBetween?.let { createdBetween(createdBetween.from, createdBetween.to) },
```

Comment:
This will only get the notes created between these dates, not the amendments. It will also only send updated events. I think it needs to send created events for the notes created in those times then find any amendments created between the two dates and send updated events for those

Reply:
The event being an update rather than a create may not matter for Delius but I think the amendment retrieval is